core/common/src/ViewProps.ts

```diff
@@ -27,6 +29,62 @@ export interface SectionDrawingViewProps {
   drawingToSpatialTransform?: TransformProps;
 }
 
+/** The response props from the getCustomViewState3dData RPC endpoint
+ * @internal
+ */
+export interface CustomViewState3dProps {
+  modelIds: CompressedId64Set;
+  modelExtents: Range3dProps;
+  categoryIds: CompressedId64Set;
+}
+
+/**
+ * The options passed to the getCustomViewState3dData RPC endpoint.
+ * @internal
+ */
+export interface CustomViewState3dCreatorOptions {
+  modelIds?: CompressedId64Set;
+}
+
+/**
+ * A result row from querying for subcategories during hydrateViewState
+ * @internal
+ */
+export interface SubCategoryResultRow {
+  parentId: Id64String;
+  id: Id64String;
+  appearance: SubCategoryAppearance.Props;
+}
+
+/**
+ * Request props for the hydrateViewState RPC endpoint.
+ * @internal
+ */
+export interface HydrateViewStateRequestProps {
+  acsId?: string;
+  notLoadedModelSelectorStateModels?: CompressedId64Set;
+  notLoadedCategoryIds?: CompressedId64Set;
+  sheetViewAttachmentIds?: CompressedId64Set;
+  sheetViewViewStateLoadProps?: ViewStateLoadProps;
+  baseModelId?: Id64String;
+  spatialViewId?: Id64String;
+  spatialViewViewStateLoadProps?: ViewStateLoadProps;
+}
+
+/** Response props from the hydrateViewState RPC endpoint.
+ * @internal
+ */
+export interface HydrateViewStateResponseProps {
+  acsElementProps?: ElementProps;
+  modelSelectorStateModels?: ModelProps[];
+  // cast this to viewattachmentInfo[] on the frontend.
```

Comment:
That would be an invalid cast - you don't actually do that right?

Reply:
I do cast it to ViewAttachmentInfo. The extra property that ViewAttachmentInfo has, "attachedView" is added shortly after. This is similar to what was taking place in the code before my changes I believe. 
Heres a screenshot of the code from SheetViewState.ts 
![image](https://user-images.githubusercontent.com/22119573/159763040-891662d8-f32b-440e-912d-691521804e4a.png)